ui/src/app/components/cases/test-case-details.component.ts

```diff
@@ -239,6 +239,43 @@ export class TestCaseDetailsComponent extends BaseComponent implements OnInit {
       }
     }
   }
+  public fetchLinkedTestCases(stepGroupId) {
+    let testCases: InfiniteScrollableDataSource;
+    testCases = new InfiniteScrollableDataSource(this.testCaseService, "workspaceVersionId:" + this.version.id + ",deleted:false,stepGroupId:" + stepGroupId);
+    waitTillRequestResponds();
+    let _this = this;
+
+    function waitTillRequestResponds() {
+      if (testCases.isFetching)
+        setTimeout(() => waitTillRequestResponds(), 100);
```

Comment:
why do we need 100 ms of timeout handler?

Reply:
if timeout is not present, then for a list of long testcases, it will not return the whole list. This handler is used in cloud also,

Reply:
promise handler won't workout here?

Reply:
im not sure bro, i asked vigneshwaran bro as he added this timeout handler in cloud and he said that it was kept so that if the list is long then it will be useful

Reply:
seems we have similar handler in multiple places, plz try to reuse the behaviour